bpf/flows.c

```diff
@@ -137,12 +123,13 @@ static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
             .dns_record.flags = pkt.dns_flags,
             .dns_record.latency = pkt.dns_latency,
             .dns_record.errno = dns_errno,
-            .flow_rtt = rtt,
         };
+        __builtin_memcpy(new_flow.dst_mac, pkt.dst_mac, ETH_ALEN);
+        __builtin_memcpy(new_flow.src_mac, pkt.src_mac, ETH_ALEN);
 
         // even if we know that the entry is new, another CPU might be concurrently inserting a flow
         // so we need to specify BPF_ANY
-        long ret = bpf_map_update_elem(&aggregated_flows, &id, &new_flow, BPF_ANY);
+        long ret = bpf_map_update_elem(&aggregated_flows, &id, &new_flow, BPF_NOEXIST);
```

Comment:
we will probably need to spin lock/unlock here as well

Reply:
the lock is on the value object, not the key, and it's a new object in that case, so I don't think it makes sense to lock

Reply:
That's what the `BPF_NOEXIST` flag does: If two CPUs call this concurrently, one of them will fail with EEXIST, and then fall back to the lookup/update path (which does take the lock)

Reply:
with eexist lookup and update we need to take the lock as well what about `BPF_F_LOCK` can I set that flag to takecare of the lock/unlock for me ?

Reply:
`BPF_F_LOCK` is for when you want to replace the whole contents of the map value; but that's not what you're doing, you are increasing a couple of counters. So you just re-use exactly the same update code as for when the map lookup succeeds in the first place :)